packages/syft/src/syft/store/mongo_document_store.py

```diff
@@ -211,27 +231,60 @@ def _set(
     ) -> Result[SyftObject, str]:
         write_permission = ActionObjectWRITE(uid=obj.id, credentials=credentials)
         can_write = self.has_permission(write_permission)
+
+        store_query_key: QueryKey = self.settings.store_key.with_obj(obj)
+        collection_status = self.collection
+        if collection_status.is_err():
+            return collection_status
+        collection: MongoCollection = collection_status.ok()
+
+        store_key_exists = (
+            collection.find_one(store_query_key.as_dict_mongo) is not None
+        )
+        if (not store_key_exists) and (not self.item_keys_exist(obj, collection)):
+            # attempt to claim ownership for writing
+            ownership_result = self.take_ownership(uid=obj.id, credentials=credentials)
+            can_write: bool = ownership_result.is_ok()
+        elif not ignore_duplicates:
+            unique_query_keys: QueryKeys = self.settings.unique_keys.with_obj(obj)
+            keys = ", ".join(f"`{key.key}`" for key in unique_query_keys.all)
```

Comment:
I'm fine with this for now, but this is a redundant check as we already create unique indexes in Mongo against unique keys (`_create_update_index`), which should already take care of this logic.